pennylane/resource/resource.py

```diff
@@ -380,6 +381,83 @@ def mul_in_parallel(r1: Resources, scalar: int) -> Resources:
     return Resources(new_wires, new_gates, new_gate_types, new_gate_sizes, r1.depth, new_shots)
 
 
+def substitute(initial_resources: Resources, gate_info: Tuple[str, int], replacement: Resources):
+    """Replaces a specified gate in a :class:`~.resource.Resources` object with the contents of another :class:`~.resource.Resources` object.
+
+    Args:
+        initial_resources (Resources): the Resource object to be modified
+        gate_info (Tuple(str, int)): tuple containing the name of the gate to be replaced and the number of wires it acts on
```

Comment:
Can this also be a list?

Reply:
Replaced with Iterable to be more specific 👍🏼